index.html

```diff
@@ -4259,6 +4239,99 @@ <h2>
           </li>
         </ol>
       </section>
+      <section>
+        <h2>
+          Can make payment algorithm
+        </h2>
+        <p>
+          The <dfn>can make payment algorithm</dfn> checks if the <a>user
+          agent</a> supports making payment with a <a>PaymentRequest</a>. It
+          takes a boolean argument, <var>checkForInstruments</var>, that
+          specifies whether the algorithm should check for existence of enrolled
+          instruments in addition to basic support of a <a>payment method</a>.
+        </p>
+        <ol class="algorithm">
+          <li>Let <var>request</var> be the <a>PaymentRequest</a> object on
+          which the method was called.
+          </li>
+          <li>If <var>request</var>.<a>[[\state]]</a> is not "<a>created</a>",
+          then return <a>a promise rejected with</a> an
+          "<a>InvalidStateError</a>" <a>DOMException</a>.
+          </li>
+          <li data-tests=
+          "payment-request/payment-request-hasenrolledinstrument-method-protection.https.html">
+          If <var>checkForInstruments</var> is true, optionally, at the
+          <a>top-level browsing context</a>'s discretion, return <a>a promise
+          rejected with</a> a "<a>NotAllowedError</a>" <a> DOMException</a>.
+            <p class="note" data-link-for="PaymentRequest">
+              This allows user agents to apply heuristics to detect and prevent
+              abuse of the <a>hasEnrolledInstrument()</a> method for
+              fingerprinting purposes, such as creating <a>PaymentRequest</a>
+              objects with a variety of supported <a>payment methods</a> and
+              calling <a>hasEnrolledInstrument()</a> on them one after the
+              other. For example, a user agent may restrict the number of
+              successful calls that can be made based on the <a>top-level
+              browsing context</a> or the time period in which those calls were
+              made.
+            </p>
+          </li>
+          <li>Let <var>promise</var> be <a>a new promise</a>.
+          </li>
+          <li>Return <var>promise</var>, and perform the remaining
+          steps <a>in parallel</a>.
+          </li>
+          <li>For each <var>paymentMethod</var> tuple in
+          <var>request</var>.<a>[[\serializedMethodData]]</a>:
+            <ol>
+              <li>Let <var>identifier</var> be the first element in the
+              <var>paymentMethod</var> tuple.
+              </li>
+              <li>If <var>checkForInstruments</var> is false, resolve <var>
+              promise</var> with true and terminate this algorithm if either the
+              user agent has a <a>payment handler</a> that supports handling
+              payment requests for <var>identifier</var>, or if it can perform
```

Comment:
I don't think we should mention the "just-in-time" stuff... that's a Chrome implementation detail that falls under "supports".

Reply:
Makes sense. Removed the JIT language.